File: Abstraction/src/ayhan/service/EmployeeService.java
package ayhan.service;

import ayhan.model.Employee;
import ayhan.util.SalaryCalculation;

public class EmployeeService extends SalaryCalculation {
    @Override
    public double calculateSalary(int workExperience,double defaultsalary) {
        if(workExperience>6){
            return defaultsalary*2;

        }
        return defaultsalary;
    }
    public Employee getEmployeeInfo(String name,
                                    int age,
                                    int workExperience,
                                    double defalultsalary){
        Employee employee=new Employee();
        employee.setName(name);
        employee.setAge(age);
        employee.setWorkExperience(workExperience);
        double empSalary=calculateSalary(workExperience,defalultsalary);
         employee.setSalary(empSalary);
        return employee;
    }
}

File: src/main/kotlin/service/EmployeeService.kt
package service

import model.Employee
import util.SalaryCalculation

class EmployeeService : SalaryCalculation() {
    override fun calculateSalary(workExperience: Int, defaultSalary: Double): Double {
        return if (workExperience > 6) {
            defaultSalary * 2
        } else defaultSalary
    }

    fun getEmployeeInfo(
        name: String?, age: Int, workExperience: Int, defaultSalary: Double
    ): Employee {
        val employee = Employee(name, age, defaultSalary, workExperience)
        val empSalary = calculateSalary(workExperience, defaultSalary)
        employee.salary = empSalary
        return employee
    }
}
